File: java/hole3/src/test/java/hole3/TakeHomeCalculatorShould.java
package hole3;


import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

@DisplayName("Hole 3: Take home calculator should")
public class TakeHomeCalculatorShould {

    @Test
    public void calculate_tax() {
        Integer first = new TakeHomeCalculator(10).netAmount(
                new TakeHomeCalculator.Money(40, "GBP"),
                new TakeHomeCalculator.Money(50, "GBP"),
                new TakeHomeCalculator.Money(60, "GBP")
        ).value;
        assertEquals(135, first.intValue());
    }

    @Test
    public void not_sum_different_currencies() {
        assertThrows(Incalculable.class, () ->
                new TakeHomeCalculator(10).netAmount(
                        new TakeHomeCalculator.Money(4, "GBP"),
                        new TakeHomeCalculator.Money(5, "USD")
                )
        );
    }
}

File: kotlin/hole1/src/test/kotlin/TakeHomeCalculatorShould.kt
package hole1

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

@DisplayName("Hole 1: Take home calculator should")
class TakeHomeCalculatorShould {

    @Test
    fun `calculate tax`() {
        val first = TakeHomeCalculator(10).netAmount(
                TakeHomeCalculator.Pair(40, "GBP"),
                TakeHomeCalculator.Pair(50, "GBP"),
                TakeHomeCalculator.Pair(60, "GBP")
        ).first
        assertEquals(135, first)
    }

    @Test
    fun `not sum different currencies`() {
        assertThrows<Incalculable> {
            TakeHomeCalculator(10).netAmount(
                    TakeHomeCalculator.Pair(4, "GBP"),
                    TakeHomeCalculator.Pair(5, "USD")
            )
        }
    }
}
